src/main/java/io/stargate/sgv2/jsonapi/api/model/command/CommandResultBuilder.java

```diff
@@ -0,0 +1,142 @@
+package io.stargate.sgv2.jsonapi.api.model.command;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.stargate.sgv2.jsonapi.exception.APIException;
+import io.stargate.sgv2.jsonapi.exception.APIExceptionCommandErrorBuilder;
+import io.stargate.sgv2.jsonapi.exception.mappers.ThrowableToErrorMapper;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class CommandResultBuilder {
+
+  public enum ResponseType {
+    SINGLE_DOCUMENT,
+    MULTI_DOCUMENT,
+    STATUS_ONLY;
+  }
+
+  private final Map<CommandStatus, Object> cmdStatus = new HashMap<>();
+  private final List<CommandResult.Error> cmdErrors = new ArrayList<>();
+  private final List<JsonNode> documents = new ArrayList<>();
+  private final List<String> warnings = new ArrayList<>();
+
+  private String nextPageState = null;
+
+  private final ResponseType responseType;
+
+  // If the debug mode is enabled, errors include the errorclass
+  private final boolean debugMode;
+
+  // Flagged true to include the new error object v2
+  private final boolean useErrorObjectV2;
+
+  // Created in the Ctor
+  private final Function<APIException, CommandResult.Error> apiExceptionToError;
+
+  public CommandResultBuilder(
```

Comment:
Would it make sense to instead use per-type factory methods for improved readability?
("forSingleDoc()", "forMultiDoc()", "forStatusOnly()" or something)

Reply:
sounds resonable, can also move the  boolean useErrorObjectV2, boolean debugMode to be builder properties 

 created https://github.com/stargate/data-api/issues/1461